### PR TITLE
Add recommended postgres config

### PR DESCRIPTION
**Pull request by agnivade**
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-31392

### CODE REVIEW

File: source/deployment/cluster.rst [hunk 1]
```diff
@@ -280,6 +280,31 @@ Transparent Failover
 
 The database can be configured for High Availability and transparent failover use the existing database technologies. We recommend MySQL Clustering, Postgres Clustering, or Amazon Aurora. Database transparent failover is beyond the scope of this documentation.
 
+Recommended Configuration Settings
+``````````````````````````````````
+
+If you are using Postgres as the choice of database, we recommend the following configuration optimizations which suits the workload of a Mattermost server.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:286]:
```suggestion
If you're using Postgres as the choice of database, we recommend the following configuration optimizations on your Mattermost server.
```

File: source/deployment/cluster.rst [hunk 2]
```diff
@@ -280,6 +280,31 @@ Transparent Failover
...
+
+The following configuration was tested on an AWS Aurora r5.xlarge instance of Postgres 11.7.
+
+1. **max_connections**: If you are using read-replicas set reader connections to 1024, and writer connections to 256. If you are using a single instance, then only setting it to 1024 should be sufficient. If the instance of lower capacity than r5.xlarge, then set it to a lower number. Also tune the `MaxOpenConns` setting under the `SqlSettings` of Mattermost app accordingly.
+
+2. **random_page_cost**: Set it to 1.1, unless the DB is using spinning disks.
+
+3. **work_mem**: Set it to 16MB for readers, and 32MB for writers. If it's a single instance, then just 16MB should be sufficient. If the instance of lower capacity than r5.xlarge, then set it to a lower number.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:294]:
```suggestion
3. **work_mem**: Set it to 16 MB for readers, and 32 MB for writers. If it's a single instance, then just 16 MB should be sufficient. If the instance of lower capacity than r5.xlarge, then set it to a lower number.
```

File: source/deployment/cluster.rst [hunk 3]
```diff
@@ -280,6 +280,31 @@ Transparent Failover
...
+
+4. **effective_cache_size**: Set it to 65% of total memory. For a 32GB instance, it should be 21GB.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:296]:
```suggestion
4. **effective_cache_size**: Set it to 65% of total memory. For a 32 GB instance, it should be 21 GB.
```

File: source/deployment/cluster.rst [hunk 4]
```diff
@@ -280,6 +280,31 @@ Transparent Failover
...
+
+5. **shared_buffers**: Set it to 65% of total memory. For a 32GB instance, it should be 21GB.
```
Comment (justinegeffen) [inline source/deployment/cluster.rst:298]:
```suggestion
5. **shared_buffers**: Set it to 65% of total memory. For a 32 GB instance, it should be 21 GB.
```

**Review by justinegeffen — APPROVED** [review]

File: source/deployment/cluster.rst [hunk 5]
```diff
@@ -280,6 +280,31 @@ Transparent Failover
...
+Recommended Configuration Settings
+``````````````````````````````````
+
+If you're using Postgres as the choice of database, we recommend the following configuration optimizations on your Mattermost server.
+
+The following configuration was tested on an AWS Aurora r5.xlarge instance of Postgres 11.7.
+
+1. **max_connections**: If you are using read-replicas set reader connections to 1024, and writer connections to 256. If you are using a single instance, then only setting it to 1024 should be sufficient. If the instance of lower capacity than r5.xlarge, then set it to a lower number. Also tune the `MaxOpenConns` setting under the `SqlSettings` of Mattermost app accordingly.
+
+2. **random_page_cost**: Set it to 1.1, unless the DB is using spinning disks.
```
Comment (streamer45) [inline source/deployment/cluster.rst:292]: Is this the same as saying, "do this if the DB is running on a solid state disk"?

**Review by streamer45 — APPROVED** [review]
Looks very good, brilliant work on these :+1: 

My only question would be how confident are we that there are no unwanted side effects. In other words, before recommending values maybe we should have a testing period (e.g. a couple of weeks).

File: source/deployment/cluster.rst (hunk 5 above)
Comment (agnivade) [reply source/deployment/cluster.rst:292]: Lol yea :)

**Comment by agnivade** [comment]
> My only question would be how confident are we that there are no unwanted side effects. In other words, before recommending values maybe we should have a testing period (e.g. a couple of weeks).

I have tested each setting individually and verified that it does what it's supposed to do, as you have already seen in the cloud-engineering channel. All these are very basic and safe to modify IMO. I have erred on the side of caution and tried to be as conservative as possible.

Although I think adding a line about the issues with cargo-culting won't hurt ;)

Totally fine with a testing period though, can you clarify what you mean by this? Who will do the testing? And how?

**Comment by streamer45** [comment]
> I have tested each setting individually and verified that it does what it's supposed to do, as you have already seen in the cloud-engineering channel. All these are very basic and safe to modify IMO. I have erred on the side of caution and tried to be as conservative as possible.

Absolutely :+1: 

> Totally fine with a testing period though, can you clarify what you mean by this? Who will do the testing? And how?

I was thinking Cloud would be start applying these setting soon so that would be more than sufficient.

**Comment by agnivade** [comment]
> I was thinking Cloud would be start applying these setting soon so that would be more than sufficient.

Ah I see. Sure, let's wait. Although it might take some time for Cloud to set all of these.

**Comment by agnivade** [comment]
So to confirm, you are good with merging this. But hold off from recommending this to our customers right?

**Comment by streamer45** [comment]
> So to confirm, you are good with merging this. But hold off from recommending this to our customers right?

Yes that works :+1:

**Pull request merged**